corehq/apps/app_manager/views/download.py

```diff
@@ -238,23 +238,28 @@ def resolve_path(path):
         return RegexURLResolver(
             r'^', 'corehq.apps.app_manager.download_urls').resolve(path)
 
+    def create_build_files_if_necessary_handling_conflicts(is_retry=False):
+        try:
+            try:
+                # look for file guaranteed to exist if profile is created
+                request.app.fetch_attachment('files/{id}/profile.xml'.format(id=build_profile))
+            except ResourceNotFound:
+                request.app.create_build_files(save=True, build_profile_id=build_profile)
+                request.app.save()
+        except ResourceConflict:
+            if is_retry:
+                raise
+            create_build_files_if_necessary_handling_conflicts(True)
+
     try:
         assert request.app.copy_of
         # lazily create language profiles to avoid slowing initial build
         try:
             payload = request.app.fetch_attachment(full_path)
         except ResourceNotFound:
             if build_profile in request.app.build_profiles and build_profile_access:
-                try:
-                    # look for file guaranteed to exist if profile is created
-                    request.app.fetch_attachment('files/{id}/profile.xml'.format(id=build_profile))
-                except ResourceNotFound:
-                    request.app.create_build_files(save=True, build_profile_id=build_profile)
-                    request.app.save()
-                    payload = request.app.fetch_attachment(full_path)
-                else:
-                    # if profile.xml is found the profile has been built and its a bad request
-                    raise
```

Comment:
Is it ok that this isn't handled anymore?

Reply:
If this situation arises the `ResourceNotFound` error will still get raised from line 262 (2 lines down). It will just be one extra request to Riak that we could have avoided but I don't think that's an issue since it's a real edge case